src/Order.js

```diff
@@ -0,0 +1,20 @@
+import React, { useState } from "react";
+import RestaurantButton from "./RestaurantButton";
+
+const Order = props => {
+  const [orders, setOrders] = useState(0);
+
+  function orderOne() {
+    setOrders(preOrder => preOrder + 1);
```

Comment:
Nice!
I dont know if we covered this, could you explain why we do this? 😄

Reply:
Because it’s a way that is safe and guaranteed to be synchronous, this is important because updating the state in a synchronous way ensures that the updated value is immediately available for other components that depend on it, which is not guaranteed when you update a variable directly.

you covered this in week 2 before exercise D 😄

Reply:
It's mentioned by Mr. Moe Ibrahim on his session on 03 Feb when we discussed the different options for state function expression, but it's not covered in the syllabus.